src/hivpy/hiv_status.py

```diff
@@ -64,12 +62,31 @@ def set_dummy_viral_load(self, population):
         """Dummy function to set viral load until this part of the code has been implemented properly"""
         population.data[col.VIRAL_LOAD_GROUP] = rng.choice(7, population.size)
 
+    def get_infection_prob(self, sex, age, n_partners, stp_age_groups):
+        # Slow example that avoid repeating the iterations over partners three time by putting them as part of 
+        # one for loop, but for loops in python will be slow.
+        target_sex = opposite_sex(sex)
+        infection_prob = np.zeros(n_partners)
+        for i in range (n_partners):
+            stp_viral_group = rng.choice(7, p = self.stp_viral_group_rate[target_sex][stp_age_groups[i]])
+            HIV_probability = self.stp_HIV_rate[opposite_sex(target_sex)][stp_age_groups[i]]
+            infection_prob[i] = HIV_probability * max(0, rng.normal(self.transmission_means[stp_viral_group], self.transmission_sigmas[stp_viral_group]))
+            if (sex == SexType.Female):
+                if(age < 20):
+                    infection_prob[i] *= self.fold_change_yw
+                else:
+                    infection_prob[i] *= self.fold_change_w
+            return infection_prob
```

Comment:
This should be out of the loop? (currently returning after computing probability for the first partner only?)

Reply:
Yes, it should be out of the loop !